internal/info/info_test.go

```diff
@@ -148,207 +155,382 @@ func TestGet(t *testing.T) {
+		afterFunc  func(args)
 	}
-	defaultCheckFunc := func(w want, got string) error {
-		if !reflect.DeepEqual(got, w.want) {
-			return errors.Errorf("got: \"%#v\",\n\t\t\t\twant: \"%#v\"", got, w.want)
+	defaultCheckFunc := func(w want, got Info) error {
+		opts := []comparator.Option{
+			comparator.AllowUnexported(info{}),
+			comparator.Comparer(func(x, y sync.Once) bool {
+				return reflect.DeepEqual(x, y)
+			}),
+			comparator.Comparer(func(x, y func(skip int) (pc uintptr, file string, line int, ok bool)) bool {
+				return reflect.ValueOf(x).Pointer() == reflect.ValueOf(y).Pointer()
+			}),
+			comparator.Comparer(func(x, y func(pc uintptr) *runtime.Func) bool {
+				return reflect.ValueOf(x).Pointer() == reflect.ValueOf(y).Pointer()
+			}),
+		}
+		if diff := comparator.Diff(w.want, got, opts...); len(diff) != 0 {
+			return errors.Errorf("err: %s", diff)
 		}
 		return nil
 	}
 	tests := []test{
-		// TODO test cases
-		/*
-		   {
-		       name: "test_case_1",
-		       fields: fields {
-		           Version: "",
-		           ServerName: "",
-		           GitCommit: "",
-		           BuildTime: "",
-		           GoVersion: "",
-		           GoOS: "",
-		           GoArch: "",
-		           CGOEnabled: "",
-		           NGTVersion: "",
-		           BuildCPUInfoFlags: nil,
-		           StackTrace: nil,
-		           PrepOnce: sync.Once{},
-		       },
-		       want: want{},
-		       checkFunc: defaultCheckFunc,
-		   },
-		*/
-
-		// TODO test cases
-		/*
-		   func() test {
-		       return test {
-		           name: "test_case_2",
-		           fields: fields {
-		           Version: "",
-		           ServerName: "",
-		           GitCommit: "",
-		           BuildTime: "",
-		           GoVersion: "",
-		           GoOS: "",
-		           GoArch: "",
-		           CGOEnabled: "",
-		           NGTVersion: "",
-		           BuildCPUInfoFlags: nil,
-		           StackTrace: nil,
-		           PrepOnce: sync.Once{},
-		           },
-		           want: want{},
-		           checkFunc: defaultCheckFunc,
-		       }
-		   }(),
-		*/
+		{
+			name: "set success when all fields are empty",
+			args: args{
+				name: "gateway",
+			},
+			want: want{
+				want: &info{
+					detail: Detail{
+						GitCommit:  "gitcommit",
+						ServerName: "gateway",
+						Version:    "gitcommit",
+						BuildTime:  "1s",
+						GoVersion:  runtime.Version(),
+						GoOS:       runtime.GOOS,
+						GoArch:     runtime.GOARCH,
+						CGOEnabled: "true",
+						NGTVersion: "v1.11.6",
+						BuildCPUInfoFlags: []string{
+							"avx512f", "avx512dq",
+						},
+					},
+					rtCaller:    runtime.Caller,
+					rtFuncForPC: runtime.FuncForPC,
+					prepOnce: func() (o sync.Once) {
+						o.Do(func() {})
+						return
+					}(),
+				},
+			},
+			beforeFunc: func(args) {
+				GitCommit = "gitcommit"
+				Version = ""
+				BuildTime = "1s"
+				CGOEnabled = "true"
+				NGTVersion = "v1.11.6"
+				BuildCPUInfoFlags = "\t\tavx512f avx512dq\t"
```

Comment:
Could you please cleanup `once` object with `afterfunc` as it may affect other tests?